MapzenSDK/MZMapViewController.swift

```diff
@@ -487,8 +487,10 @@ open class MZMapViewController: UIViewController, LocationManagerDelegate {
    - parameter marker: The marker to add to map.
    */
   open func addMarker(_ marker: GenericMarker) {
-    currentMarkers[marker.tgMarker] = marker
-    marker.tgMarker.map = tgViewController
+    if marker.tgMarker == nil {
+      marker.tgMarker = tgViewController.markerAdd()
+    }
+    currentMarkers[marker.tgMarker!] = marker
```

Comment:
I actually like the `NSDictionary` idea, it would let you store different kind of data without having to create your own data structure holding it. Feel free to open an issue in ES with the different options we have.